pymc3/dp/dp.py

```diff
@@ -0,0 +1,170 @@
+#   Copyright 2021 The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+import sys
+
+sys.path.insert(0, "../../../pymc3")
+
+import numpy as np
+
+from aesara import tensor as at
+
+import pymc3 as pm
+
+# from pymc3.distributions.dist_math import bound
+# from pymc3.distributions.distribution import Distribution
+# from pymc3.dp.util import StickBreakingWeights
+
+__all__ = [
+    "DirichletProcess",
+]
+
+
+class DPBase:
```

Comment:
@AustinRochford  @fonnesbeck What are your thoughts on the `DPBase` class as created here? This was largely inspired by the `Base` [class](https://github.com/pymc-devs/pymc3/blob/main/pymc3/gp/gp.py#L46) for Gaussian Processes, so I'm wondering if I should "redesign" how I approach DPs in PyMC3.